src/OAuth2/Encryption/Jwt.php

```diff
@@ -90,24 +100,17 @@ private function verifySignature($signature, $input, $key, $algo = 'HS256')
 
     private function sign($input, $key, $algo = 'HS256')
     {
-        switch ($algo) {
-            case 'HS256':
-                return hash_hmac('sha256', $input, $key, true);
-
-            case 'HS384':
-                return hash_hmac('sha384', $input, $key, true);
-
-            case 'HS512':
-                return hash_hmac('sha512', $input, $key, true);
+        list($function, $algorithm) = self::$supportedAlgorithms[$algo];
 
-            case 'RS256':
-                return $this->generateRSASignature($input, $key, defined('OPENSSL_ALGO_SHA256') ? OPENSSL_ALGO_SHA256 : 'sha256');
+        switch ($function) {
+            case 'hash_hmac':
+                return hash_hmac($algorithm, $input, $key, true);
 
-            case 'RS384':
-                return $this->generateRSASignature($input, $key, defined('OPENSSL_ALGO_SHA384') ? OPENSSL_ALGO_SHA384 : 'sha384');
-
-            case 'RS512':
-                return $this->generateRSASignature($input, $key, defined('OPENSSL_ALGO_SHA512') ? OPENSSL_ALGO_SHA512 : 'sha512');
+            case 'openssl':
+                if(defined(self::$supportedAlgorithms[$algo][2])){
+                    $algorithm = constant(self::$supportedAlgorithms[$algo][2]);
```

Comment:
Basically yeah, it is a copy of what the `firebase/php-jwt` lib does. I realise it is an increase in complexity, and if you'd rather remove it thats all good. I just figured it wouldn't be a good idea to hardcode the supported RSA algorithms and have to maintain them separately from the original switch statement that was there. Since the different function have very similar variable signatures it's not too bad.